src/com/company/exercises/Demo.java

```diff
@@ -0,0 +1,36 @@
+package com.company.exercises;
+
+public class Demo {
+    @FunctionalInterface
+    public interface StringRunnable {
+        public abstract void run(String str);
+    }
+
+    //FI with the same function descriptor Runnable
+    @FunctionalInterface
+    public interface Executor {
+        void execute();
+    }
+
+    public static void main(String[] args) {
+// lambda with str param and no param parenthesis when defined
+
+        StringRunnable fn2 = name -> System.out.println("hello" + name + "!");
+        fn2.run("Rado");
+
+        //        simple lambdas no param
+
+        Runnable fn = () -> {
+            String name = "Matt";
+            System.out.println("hello" + name + "!");
+        };
+        fn.run();
+
+        Executor fn_ = () -> {
```

Comment:
Avoid naming stuff like this, especially the underscore.
I suggest you take a look at chapter 2 of Clean code when you get some free time. It is all about proper naming.
Let me know if you need help finding the book.

Reply:
Done.